src/hangar/cli/cli.py

```diff
@@ -359,136 +360,172 @@ def server(overwrite, ip, port, timeout):
+                    if not overwrite and fname in aset:
+                        continue
+                    try:
+                        aset[fname] = arr
+                    except ValueError as e:
+                        click.echo(e)
     finally:
         co.close()
 
 
-@main.command(name='export')
-@click.argument('startpoint', nargs=1, required=True)
+@main.command(name='export', context_settings=dict(allow_extra_args=True, ignore_unknown_options=True,))
 @click.argument('arrayset', required=True)
-@click.option('-o', '--out', required=True, help='Path to export the data to.')
-@click.option('-s', '--sample', default=False, help='Sample name to export')
-@click.option('-f', '--format', 'format_', required=False, help='File format used for exporting.')
+@click.argument('startpoint', nargs=1, default=None, required=False)
+@click.option('-o', '--out', required=False, default=os.getcwd(), help="Directory to export data")
+@click.option('-s', '--sample', default=None, help='Sample name to export. You can optionally specify the type'
+                                                   ' of sample name after the colon (ex. 54:str or 54:int)')
+@click.option('-f', '--format', 'format_', required=False, help='File format of output file')
 @click.option('--plugin', required=False, help='override auto-inferred plugin')
 @pass_repo
-def export_data(repo: Repository, startpoint, arrayset, out, sample, format_, plugin):
-    """export ARRAYSET sample data as it existed a STARTPOINT to some format and path.
+@click.pass_context
+def export_data(ctx, repo: Repository, arrayset, out, startpoint, sample, format_, plugin):
+    """
+    Export ARRAYSET sample data as it existed in the branch or commit to some format and path.
     """
+    # TODO: ignore warning through env variable
+    # TODO: document --sample exceptional case
     from hangar.records.commiting import expand_short_commit_digest
-    from hangar.records.heads import get_branch_head_commit
-    from hangar.cli.io import imsave
+    from hangar.records.heads import get_branch_head_commit, get_staging_branch_head
+    from hangar import external
+
+    kwargs = parse_custom_arguments(ctx.args)
 
     if startpoint in repo.list_branches():
         base_commit = get_branch_head_commit(repo._env.branchenv, startpoint)
-    else:
+    elif startpoint:
         base_commit = expand_short_commit_digest(repo._env.refenv, startpoint)
+    else:
+        branch_name = get_staging_branch_head(repo._env.branchenv)
+        base_commit = get_branch_head_commit(repo._env.branchenv, branch_name)
+    co = repo.checkout(commit=base_commit)
 
     try:
-        co = repo.checkout(write=False, commit=base_commit)
-        arrayset = co.arraysets[arrayset]
+        aset = co.arraysets.get(arrayset)
         if sample:
+            sample, stype = sample.split(':') if ':' in sample else (sample, '')
+            if stype == 'int' or stype == 'str':
+                sample = eval(stype)(sample)
```

Comment:
don't use eval.

```suggestion
                sample = int(sample) if stype == 'int' else str(sample)
```

Reply:
Better yet, have a `stype_to_type = {'int': int, 'str': str}` dict somewhere (maybe even somewhere plugins can add things to), and dispatch on that.

Reply:
Done